tests/unit/viahtml/stats_test.py

```diff
@@ -0,0 +1,147 @@
+        # not what.
+
+        results = list(stats())
+
+        print(results)
+
+        # To maintain these tests when things change print out results, examine
+        # them and then set them to expected if they look good
+        expected = [
+            ("Custom/Alive/Metrics", 1),
+            ("Custom/Alive/App", 1),
+            ("Custom/System/Load", 16),
+            ("Custom/ListenQueue/Size", 12),
+            ("Custom/ListenQueue/Errors", 13),
+            ("Custom/SignalQueue/Size", 14),
+            (
+                "Custom/Socket/Queue/Size",
+                {"total": 17, "count": 1, "min": 17, "max": 17, "sum_of_squares": 289},
+            ),
+            (
+                "Custom/Socket/Queue/Max",
+                {
+                    "total": 100,
+                    "count": 1,
+                    "min": 100,
+                    "max": 100,
+                    "sum_of_squares": 10000,
+                },
+            ),
+            ("Custom/Worker/Count/Accepting", 5),
+            ("Custom/Worker/Count/Max", 7),
+            (
+                "Custom/Worker/Request/Total",
+                {"total": 14, "count": 5, "min": 0, "max": 12, "sum_of_squares": 148},
+            ),
+            (
+                "Custom/Worker/Request/Failed",
+                {"total": 7, "count": 5, "min": 0, "max": 5, "sum_of_squares": 29},
+            ),
+            (
+                "Custom/Worker/Killed",
+                {"total": 7, "count": 5, "min": 0, "max": 4, "sum_of_squares": 21},
+            ),
+            (
+                "Custom/Worker/Memory/Resident[kB]",
+                {
+                    "total": 44796.48,
+                    "count": 5,
+                    "min": 3.12,
+                    "max": 44784.0,
+                    "sum_of_squares": 2005606694,
+                },
+            ),
+            (
+                "Custom/Worker/AverageResponseTime[ms]",
+                {
+                    "total": 177.04,
+                    "count": 2,
+                    "min": 0.34,
+                    "max": 176.7,
+                    "sum_of_squares": 31223,
+                },
+            ),
+            (
+                "Custom/Worker/UpTime[ms]",
+                {
+                    "total": 354.56,
+                    "count": 1,
+                    "min": 354.56,
+                    "max": 354.56,
+                    "sum_of_squares": 125712,
+                },
+            ),
+            (
+                "Custom/Worker/Memory/Virtual[kB]",
+                {
+                    "total": 748300.0,
+                    "count": 1,
+                    "min": 748300.0,
+                    "max": 748300.0,
+                    "sum_of_squares": 559952890000,
+                },
+            ),
+        ]
+
+        for metric_name, value in expected:
+            assert (metric_name, value) in results
```

Comment:
We could do this all in one go, but this at least gives you a fighting chance of spotting individual problems